xml/System/IntPtr.xml

```diff
@@ -244,23 +244,25 @@
     </Attribute>
   </Attributes>
   <Docs>
-    <summary>A platform-specific type that is used to represent a pointer or a handle.</summary>
+    <summary>Represents a signed integer where the bit-width is the same as a pointer.</summary>
     <remarks>
       <format type="text/markdown"><![CDATA[
 
 ## Remarks
 
- The <xref:System.IntPtr> type is designed to be an integer whose size is platform-specific. That is, an instance of this type is expected to be 32-bits on 32-bit hardware and operating systems, and 64-bits on 64-bit hardware and operating systems.
+ The <xref:System.IntPtr> type is designed to be an integer whose size is the same as a pointer. That is, an instance of this type is expected to be 32-bits in a 32-bit process and 64-bits in a 64-bit process.
 
  The <xref:System.IntPtr> type can be used by languages that support pointers and as a common means of referring to data between languages that do and do not support pointers.
 
  <xref:System.IntPtr> objects can also be used to hold handles. For example, instances of <xref:System.IntPtr> are used extensively in the <xref:System.IO.FileStream?displayProperty=nameWithType> class to hold file handles.
 
- The <xref:System.IntPtr> type is CLS-compliant, while the <xref:System.UIntPtr> type is not. Only the <xref:System.IntPtr> type is used in the common language runtime. The <xref:System.UIntPtr> type is provided mostly to maintain architectural symmetry with the <xref:System.IntPtr> type.
+ However, <xref:System.IntPtr> is not itself a pointer or a handle. It is simply an integer type that can be used as an interchange format for pointers and handles due to being the same size. Outside of specific interchange requirements, such as for passing data to a language that doesn't support pointers, a correctly typed pointer should be used to represent pointers and <xref:System.Runtime.InteropServices.SafeHandle> should be used to represent handles.
```

Comment:
This is likely `important`. Is there any way we can raise visibility on it and point users towards using `SafeHandle` for handles and the corresponding docs for it?

Likewise, should we elaborate at all why using `IntPtr` as a pointer type is error prone and `unsafe`? For example, converting some `int*` to `IntPtr` loses the context that it is a single indirection and the backing data is `int`. A user getting an `IntPtr` could therefore accidentally pass in an `int**` or a `byte*` and the data may end up incorrectly handled.

Reply:
You could use syntax like this:

    > [!NOTE]
    > Using IntPtr as a pointer type is error prone and unsafe.

There are more exceptions to this then only passing to a language that doesn't support pointers, though. For example Marshal.AllocHGlobal/FreeHGlobal also use IntPtr to represent pointers.